src/webview/components/memory-table.tsx

```diff
@@ -151,8 +193,29 @@ export class MemoryTable extends React.PureComponent<MemoryTableProps, MemoryTab
         };
     }
 
+    componentDidMount(): void {
+        this.resizeObserver = new ResizeObserver(entries => {
+            // The size changed - we could have too few rows visible to enable a scrollbar
+            if (entries.length > 0 && this.props.scrollingBehavior === 'Auto-Append') {
+                this.ensureSufficientVisibleRowsForScrollbar();
+            }
+        });
+
+        const element = this.datatableRef.current?.getElement();
+        if (element) {
+            this.resizeObserver.observe(element);
+        }
+    }
+
     componentDidUpdate(prevProps: Readonly<MemoryTableProps>): void {
-        const hasMemoryChanged = prevProps.memory?.address !== this.props.memory?.address || prevProps.offset !== this.props.offset || prevProps.count !== this.props.count;
+        if (!MemoryState.hasMemory(prevProps) || !MemoryState.hasMemory(this.props)) {
```

Comment:
Supposing `MemoryState.hasMemory` returns `false` for `prevProps` but `true` for the new props, would we not want to perform the update activities from lines 223-235?